CloudM.Exercise.WebApi/Controllers/PackageController.cs

```diff
@@ -27,12 +31,63 @@ public async Task<PackageDependencies> Get(string name, string version)
                 PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
             });
 
+            var dependencies = npmPackage.Versions[version].Dependencies;
+
+            foreach (var dep in dependencies)
+            {
+                var subDep = GetDependencies(dep.Key, dep.Value).Result;
+                dependencyTree.Add(dep.Key, new
+                {
+                    Version = dep.Value,
+                    Dependecies = subDep
+                });
+            }
+
             return new PackageDependencies
             {
                 Name = name,
                 Version = version,
-                Dependencies = npmPackage.Versions[version].Dependencies,
+                Dependencies = dependencyTree,
             };
         }
+
+        public async Task<Dictionary<string, dynamic>> GetDependencies(string name, string version)
```

Comment:
The method should be made private as it is only required to be used to get the dependencies in the Get request or for recursion by calling itself. In its current state the method will cause the swagger documentation to display an error as it is expecting the public method to have annotation for an HTTP method.